config/v0.3.0/propagation.go

```diff
@@ -0,0 +1,65 @@
+package config
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/contrib/propagators/aws/xray"
+	"go.opentelemetry.io/contrib/propagators/b3"
+	"go.opentelemetry.io/contrib/propagators/jaeger"
+	"go.opentelemetry.io/contrib/propagators/ot"
+	"go.opentelemetry.io/otel/propagation"
+)
+
+func propagator(cfg configOptions) (propagation.TextMapPropagator, error) {
+	if cfg.opentelemetryConfig.Propagator == nil {
+		return nil, nil
+	}
+
+	var errs []error
+	var ps []propagation.TextMapPropagator
+	for _, name := range cfg.opentelemetryConfig.Propagator.Composite {
+		if name == nil || *name == "" {
+			continue
+		}
+
+		p, err := propagatorByName(*name)
+		if err == nil {
+			ps = append(ps, p)
+		} else {
+
+			errs = append(errs, err)
+		}
+	}
+
+	if len(errs) > 0 {
+		return nil, errors.Join(errs...)
+	}
+
+	if len(ps) == 0 {
+		return nil, nil
+	}
+
+	res := propagation.NewCompositeTextMapPropagator(ps...)
+	return res, nil
+}
+
+func propagatorByName(name string) (propagation.TextMapPropagator, error) {
```

Comment:
_Not a blocker. This is likely to be tracked as a separate issue._

Should we not take advantage of https://pkg.go.dev/go.opentelemetry.io/contrib/propagators/autoprop? Similarly for exporters we could use https://pkg.go.dev/go.opentelemetry.io/contrib/exporters/autoexport.

I think that the config is supposed to be extensible and allow custom exporters, processors, propagators. See: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk.md#sdk-extension-components

Reply:
I agree. I created https://github.com/open-telemetry/opentelemetry-go-contrib/issues/6728. Resolving.